modules/renter/download.go

```diff
@@ -549,8 +611,7 @@ func (r *Renter) managedWaitOnDownloadWork(ds *downloadState) {
 	// If the chunk has completed, perform chunk recovery.
 	if len(cd.completedPieces) == cd.download.erasureCode.MinPieces() {
 		err := cd.recoverChunk()
-		ds.activePieces -= len(cd.completedPieces)
-		cd.completedPieces = make(map[uint64][]byte)
```

Comment:
To be honest, this code is not super well organized and I'm not certain whether this will cause problems or not.

The best way to tell would be an integration test that downloads:

+ a multi-chunk file with enough hosts to utilize the extra pieces
+ a multi-chunk file with seemingly enough hosts to utilize the extra pieces, except some of the hosts have gone offline so really you only have exactly enough hosts available
+ a file with more chunks than hosts that can utilize the full paralleism of the downloader

If it can do all these things, then this code is probably not worse than the existing code.

To be honest, the existing download code is pretty sloppy (my own design) and could use some design iteration. Mostly, it doesn't seem right to me that this file would need to be a whole 500 lines of code - we should be able to achieve everything it does in a lot fewer lines than that (I think, anyway).